File: src/typescript/structure/organisation/consistency/supermarket.ts
class UserService {
    fetchUserProfile(userId: number): Promise<string> {
        return new Promise((resolve) => {
            setTimeout(() => {
                resolve(`User Profile for ID: ${userId}`);
            }, 1000);
        });
    }

    checkUserEligibility(age: number): boolean {
        return age >= 18;
    }
}

class ProductService {
    retrieveProductDetail(productId: number): string {
        const products: { [key: number]: string } = {
            101: "Laptop",
            102: "Smartphone",
            103: "Headphones",
        };
        return products[productId] || "Product Not Found";
    }

    validateProductStock(stock: number): boolean {
        return stock > 0;
    }
}

class OrderService {
    getOrderInfo(orderId: number): string {
        const orders: { [key: number]: string } = {
            5001: "Order #5001: Shipped",
            5002: "Order #5002: Processing",
            5003: "Order #5003: Delivered",
        };
        return orders[orderId] || "Order Not Found";
    }

    isOrderComplete(completed: boolean): boolean {
        return completed;
    }
}


const userService = new UserService();
userService.fetchUserProfile(1).then(profile => console.log(profile));
console.log(userService.checkUserEligibility(20));

const productService = new ProductService();
console.log(productService.retrieveProductDetail(101));
console.log(productService.validateProductStock(0));

const orderService = new OrderService();
console.log(orderService.getOrderInfo(5002));
console.log(orderService.isOrderComplete(true));
\ No newline at end of file

File: truth_set/go/structure/organisation/consistency/supermarket.go
package supermarket

import (
	"fmt"
	"time"
)

type UserService struct{}

func (s *UserService) GetUserProfile(userID int) string {
	time.Sleep(time.Second)
	return fmt.Sprintf("User Profile for ID: %d", userID)
}

func (s *UserService) CheckUserEligibility(age int) bool {
	return age >= 18
}

type ProductService struct{}

func (s *ProductService) GetProduct(productID int) string {
	products := map[int]string{
		101: "Laptop",
		102: "Smartphone", 
		103: "Headphones",
	}
	if product, ok := products[productID]; ok {
		return product
	}
	return "Product Not Found"
}

func (s *ProductService) ValidateProductStock(stock int) bool {
	return stock > 0
}

type OrderService struct{}

func (s *OrderService) RetrieveOrder(orderID int) string {
	orders := map[int]string{
		5001: "Order #5001: Shipped",
		5002: "Order #5002: Processing",
		5003: "Order #5003: Delivered",
	}
	if order, ok := orders[orderID]; ok {
		return order
	}
	return "Order Not Found"
}

func (s *OrderService) VerifyOrderStatus(completed bool) bool {
	return completed
}

func main() {
	userService := &UserService{}
	fmt.Println(userService.GetUserProfile(1))
	fmt.Println(userService.CheckUserEligibility(20))

	productService := &ProductService{}
	fmt.Println(productService.GetProduct(101))
	fmt.Println(productService.ValidateProductStock(0))

	orderService := &OrderService{}
	fmt.Println(orderService.RetrieveOrder(5002))
	fmt.Println(orderService.VerifyOrderStatus(true))
}
\ No newline at end of file